File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.cs
public class Solution {
    public int MinOperations(int[] a, int k) 
    {
        int ans = 0;
        PriorityQueue<long, long> Q = new PriorityQueue<long, long>();
        foreach (int i in a)
            Q.Enqueue(i, i);

        while(Q.Count > 1)
        {
            long first = Q.Peek();
            Q.Dequeue();

            if (first >= k)
                break;
            long second = Q.Peek();
            Q.Dequeue();
            long newNumber = 2*first + second;
            Q.Enqueue(newNumber, newNumber);
            ans++;
        }

        return ans;
    }
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) 
    {
        priority_queue <long, vector<long>, greater<long>> q;
        int ans = 0;
        for (int i : nums) 
            q.push(i);
        
        while(q.size() > 1)
        {
            long first = q.top();
            q.pop();
            if (first >= k)
                break;
            long second = q.top();
            q.pop();
            long newNumber = first*2 + second;
            q.push(newNumber);
            ans++;
        }
        return ans;
    }
};